test/testdata/rewriter/initializer.rb

```diff
@@ -46,6 +46,35 @@ def foo
   end
 end
 
+class Literals
+  extend T::Sig
+
+  sig {void}
+  def initialize
+    @int = 0
+    @str = 'str'
+    @sym = :sym
+    @float = 0.0
+    @nil = nil
+    @true = true
+    @false = false
+    @cls = Object
+    @obj = Object.new
+  end
+
+  sig {void}
+  def foo
+    T.reveal_type(@int) # error: Revealed type: `Integer`
+    T.reveal_type(@str) # error: Revealed type: `String`
+    T.reveal_type(@sym) # error: Revealed type: `Symbol`
+    T.reveal_type(@float) # error: Revealed type: `Float`
+    T.reveal_type(@nil) # error: Revealed type: `NilClass`
+    T.reveal_type(@true) # error: Revealed type: `TrueClass`
```

Comment:
Way out of scope, but IMHO we should change type inference globally so that we always infer `T::Boolean` from bare literal `true` and `false` values instead of inferring the too-specific `TrueClass` or `FalseClass`; how often do you actually want a boolean that can only ever have one value?

Reply:
https://github.com/sorbet/sorbet/blob/master/test/testdata/infer/true_false_knowledge.rb#L3-L15

If you can get it to work in a way that doesn't break this test case, then be my guest. I think that this is unfortunately going to be a limitation of Sorbet's inference algorithm for a while.

Reply:
Oh ugh. I don't know anything about compilers really but that does seem like something where it'd be hard to get the best of both worlds without doing something way fancier than the AST rewriter for type inference.

Reply:
(Appreciate getting the chance to learn a bit about this stuff thanks to the good-first-issue tag!)